commander/tracker/txs_tracker.go

```diff
@@ -0,0 +1,71 @@
+package tracker
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Worldcoin/hubble-commander/eth"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type TxsTracker struct {
+	TxsChan chan *types.Transaction
+	*TxsSender
+}
+
+func NewTxTracker(ethClient *eth.Client, txsChan chan *types.Transaction) *TxsTracker {
+	return &TxsTracker{
+		TxsSender: newTxRequestsSender(ethClient),
+		TxsChan:   txsChan,
+	}
+}
+
+func (t *TxsTracker) StartTracking(ctx context.Context) error {
+	sentTxs := make([]*types.Transaction, 0)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case reqParams := <-t.requestsChan:
+			_, err := t.sendRequest(reqParams)
+			if err != nil {
+				panic(err)
+			}
+			continue
+		case tx := <-t.TxsChan:
+			sentTxs = append(sentTxs, tx)
+			continue
+		default:
+			break
+		}
+
+		if len(sentTxs) == 0 {
+			time.Sleep(time.Millisecond * 100)
+			continue
+		}
+
+		isMined, receipt, err := t.client.IsTxMined(sentTxs[0])
+		if err != nil {
+			panic(err)
+		}
+		if !isMined {
+			time.Sleep(time.Millisecond * 100)
+			continue
+		}
+		err = t.checkTxForFail(sentTxs[0], receipt)
+		if err != nil {
+			panic(err)
+		}
+		sentTxs = sentTxs[1:]
```

Comment:
It's look like `WaitToBeMined` with additional `checkTxForFail` function and . I would reuse that function, especially that this logic shouldn't block receiving sent txs. I would call function with `WaitToBeMined` in separate goroutine with first tx and use some kind of error channel to return errors from this function and once it's done we should call that function with next tx.

Reply:
I split the tracker into two workers, and now creating additional goroutines makes no sense.